src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -259,7 +264,15 @@ def _normalize_properties(self, user, subscriptions, is_service_principal, cert_
                 _IS_DEFAULT_SUBSCRIPTION: False,
                 _TENANT_ID: s.tenant_id,
                 _ENVIRONMENT_NAME: self.cli_ctx.cloud.name
-            })
+            }
+            # for Subscriptions - List REST API 2019-06-01's subscription account
+            if subscription_dict[_SUBSCRIPTION_NAME] != _TENANT_LEVEL_ACCOUNT_NAME:
```

Comment:
>_TENANT_LEVEL_ACCOUNT_NAME [](start = 56, length = 26)

If a subscription is a tenant level account, does it mean it only belong to one tenant? In this case, the rest api response does not has homeTenentId? why do we need this check here?

Reply:
> If a subscription is a tenant level account, does it mean it only belong to one tenant? 

In that case, the account **is** a tenant with no subscriptions, not "belong to". The naming of `subscription_dict` is inaccurate. It contains both subscription and tenant accounts. 

> In this case, the rest api response does not has homeTenentId? 

This tenant account is not generated from REST, but manually built up:

https://github.com/Azure/azure-cli/blob/944c0c3b1f30530a280bfaa6b7c7d799e3248f28/src/azure-cli-core/azure/cli/core/_profile.py#L270-L279

> Why do we need this check here?

It doesn't make sense to include `home_tenant_id` and `managed_by_tenants` (whichs are subscription account's attributes) for a tenant account.

In fact, this is all due to the faulty original design that mixes subscription and tenant in the same list. This will and must be changed in the future to reflect the hierarchy. Tenant is a one-level-higher concept than subscription.